src/drivers/batt_smbus/batt_smbus.cpp

```diff
@@ -602,46 +514,17 @@ BATT_SMBUS::cycle()
 			}
 		}
 
-		// if it is a solo battery, check for shutdown on button press
-		if (_is_solo_battery) {
-			// read the button press indicator
-			if (read_block(BATT_SMBUS_MANUFACTURER_DATA, buff, 6, false) == 6) {
-				bool pressed = (buff[1] >> 3) & 0x01;
-
-				if (_button_press_counts >= ((BATT_SMBUS_BUTTON_DEBOUNCE_MS * 1000) / BATT_SMBUS_MEASUREMENT_INTERVAL_US)) {
-					// battery will power off
-					new_report.is_powering_off = true;
-
-					// warn only once
-					if (_button_press_counts++ == ((BATT_SMBUS_BUTTON_DEBOUNCE_MS * 1000) / BATT_SMBUS_MEASUREMENT_INTERVAL_US)) {
-						PX4_WARN("system is shutting down NOW...");
-					}
-
-				} else if (pressed) {
-					// battery will power off if the button is held
-					_button_press_counts++;
-
-				} else {
-					// button released early, reset counters
-					_button_press_counts = 0;
-					new_report.is_powering_off = false;
-				}
-			}
-		}
-
-
 		// publish to orb
-		if (_batt_topic != nullptr) {
-			orb_publish(_batt_orb_id, _batt_topic, &new_report);
+		actuator_controls_s ctrl;
+		orb_copy(ORB_ID(actuator_controls_0), _actuator_ctrl_0_sub, &ctrl);
 
-		} else {
-			_batt_topic = orb_advertise(_batt_orb_id, &new_report);
-
-			if (_batt_topic == nullptr) {
-				PX4_ERR("ADVERT FAIL");
-				return;
-			}
-		}
+		battery_status_s battery_status;
+		_battery.updateBatteryStatus(now, new_report.voltage_v, new_report.current_a,
```

Comment:
I see why you did this, but I really think we need to find a better overall solution than having a low level battery driver subscribing to controller output like this.

Reply:
I agree, the sensors.cpp battery update process also relies on it.